### ISSUE TITLE
`callback` for saving intermediate iterates?

### ISSUE THREAD

**Issue by aringh** [odlgroup/odl#639]
We have `callback`s for showing intermediate iterations, but is there an easy way to save these plots to disc?

There is a `callback` for applying functions to intermediate iterates, and I assume that one could write some kind of `lambda`-function for it there. Do we want a "standard implementation" of that in odl?

**Comment by adler-j** [comment]
There is no standard way but it is almost trivially done. Freestyle coded example here:

```python
class CallbackShowAndSave(Callback):
    def __init__(self, file_prefix, **kwargs):
        self.iter = 0
        self.file_prefix = file_prefix
        self.kwargs = kwargs

    def __call__(self, x):
        x.show(**self.kwargs, saveto=os.path.join(self.file_prefix, '_' + str(self.iter))
        self.iter += 1
```

An option, as you note, would certainly be to add a option to the `CallbackShow` for doing this.

**Comment by aringh** [comment]
Closed by PR #708
